diff -u -r nacl-20100830.orig/crypto_onetimeauth/poly1305/amd64/auth.s nacl-20100830/crypto_onetimeauth/poly1305/amd64/auth.s
--- nacl-20100830.orig/crypto_onetimeauth/poly1305/amd64/auth.s	2010-08-30 04:25:19.000000000 +0200
+++ nacl-20100830/crypto_onetimeauth/poly1305/amd64/auth.s	2011-01-07 23:54:07.000000000 +0100
@@ -2785,3 +2785,86 @@
 xor %rax,%rax
 xor %rdx,%rdx
 ret
+# version 20080913
+# D. J. Bernstein
+# Public domain.
+
+.p2align 5
+
+#.globl _crypto_onetimeauth_poly1305_amd64_constants
+#.globl crypto_onetimeauth_poly1305_amd64_constants
+#.globl crypto_onetimeauth_poly1305_amd64_scale
+#.globl crypto_onetimeauth_poly1305_amd64_two32
+#.globl crypto_onetimeauth_poly1305_amd64_two64
+#.globl crypto_onetimeauth_poly1305_amd64_two96
+#.globl crypto_onetimeauth_poly1305_amd64_alpha32
+#.globl crypto_onetimeauth_poly1305_amd64_alpha64
+#.globl crypto_onetimeauth_poly1305_amd64_alpha96
+#.globl crypto_onetimeauth_poly1305_amd64_alpha130
+#.globl crypto_onetimeauth_poly1305_amd64_doffset0
+#.globl crypto_onetimeauth_poly1305_amd64_doffset1
+#.globl crypto_onetimeauth_poly1305_amd64_doffset2
+#.globl crypto_onetimeauth_poly1305_amd64_doffset3
+#.globl crypto_onetimeauth_poly1305_amd64_doffset3minustwo128
+#.globl crypto_onetimeauth_poly1305_amd64_hoffset0
+#.globl crypto_onetimeauth_poly1305_amd64_hoffset1
+#.globl crypto_onetimeauth_poly1305_amd64_hoffset2
+#.globl crypto_onetimeauth_poly1305_amd64_hoffset3
+#.globl crypto_onetimeauth_poly1305_amd64_rounding
+
+_crypto_onetimeauth_poly1305_amd64_constants:
+crypto_onetimeauth_poly1305_amd64_constants:
+crypto_onetimeauth_poly1305_amd64_scale:
+.long 0x0,0x37f40000
+
+crypto_onetimeauth_poly1305_amd64_two32:
+.long 0x0,0x41f00000
+
+crypto_onetimeauth_poly1305_amd64_two64:
+.long 0x0,0x43f00000
+
+crypto_onetimeauth_poly1305_amd64_two96:
+.long 0x0,0x45f00000
+
+crypto_onetimeauth_poly1305_amd64_alpha32:
+.long 0x0,0x45e80000
+
+crypto_onetimeauth_poly1305_amd64_alpha64:
+.long 0x0,0x47e80000
+
+crypto_onetimeauth_poly1305_amd64_alpha96:
+.long 0x0,0x49e80000
+
+crypto_onetimeauth_poly1305_amd64_alpha130:
+.long 0x0,0x4c080000
+
+crypto_onetimeauth_poly1305_amd64_doffset0:
+.long 0x0,0x43300000
+
+crypto_onetimeauth_poly1305_amd64_doffset1:
+.long 0x0,0x45300000
+
+crypto_onetimeauth_poly1305_amd64_doffset2:
+.long 0x0,0x47300000
+
+crypto_onetimeauth_poly1305_amd64_doffset3:
+.long 0x0,0x49300000
+
+crypto_onetimeauth_poly1305_amd64_doffset3minustwo128:
+.long 0x0,0x492ffffe
+
+crypto_onetimeauth_poly1305_amd64_hoffset0:
+.long 0xfffffffb,0x43300001
+
+crypto_onetimeauth_poly1305_amd64_hoffset1:
+.long 0xfffffffe,0x45300001
+
+crypto_onetimeauth_poly1305_amd64_hoffset2:
+.long 0xfffffffe,0x47300001
+
+crypto_onetimeauth_poly1305_amd64_hoffset3:
+.long 0xfffffffe,0x49300003
+
+crypto_onetimeauth_poly1305_amd64_rounding:
+.byte 0x7f
+.byte 0x13
Nur in nacl-20100830/crypto_onetimeauth/poly1305/amd64: auth.s~.
diff -u -r nacl-20100830.orig/crypto_onetimeauth/poly1305/amd64/constants.s nacl-20100830/crypto_onetimeauth/poly1305/amd64/constants.s
--- nacl-20100830.orig/crypto_onetimeauth/poly1305/amd64/constants.s	2010-08-30 04:25:19.000000000 +0200
+++ nacl-20100830/crypto_onetimeauth/poly1305/amd64/constants.s	2011-01-07 22:44:37.000000000 +0100
@@ -1,85 +0,0 @@
-# version 20080913
-# D. J. Bernstein
-# Public domain.
-
-.data
-.section .rodata
-.p2align 5
-
-.globl _crypto_onetimeauth_poly1305_amd64_constants
-.globl crypto_onetimeauth_poly1305_amd64_constants
-.globl crypto_onetimeauth_poly1305_amd64_scale
-.globl crypto_onetimeauth_poly1305_amd64_two32
-.globl crypto_onetimeauth_poly1305_amd64_two64
-.globl crypto_onetimeauth_poly1305_amd64_two96
-.globl crypto_onetimeauth_poly1305_amd64_alpha32
-.globl crypto_onetimeauth_poly1305_amd64_alpha64
-.globl crypto_onetimeauth_poly1305_amd64_alpha96
-.globl crypto_onetimeauth_poly1305_amd64_alpha130
-.globl crypto_onetimeauth_poly1305_amd64_doffset0
-.globl crypto_onetimeauth_poly1305_amd64_doffset1
-.globl crypto_onetimeauth_poly1305_amd64_doffset2
-.globl crypto_onetimeauth_poly1305_amd64_doffset3
-.globl crypto_onetimeauth_poly1305_amd64_doffset3minustwo128
-.globl crypto_onetimeauth_poly1305_amd64_hoffset0
-.globl crypto_onetimeauth_poly1305_amd64_hoffset1
-.globl crypto_onetimeauth_poly1305_amd64_hoffset2
-.globl crypto_onetimeauth_poly1305_amd64_hoffset3
-.globl crypto_onetimeauth_poly1305_amd64_rounding
-
-_crypto_onetimeauth_poly1305_amd64_constants:
-crypto_onetimeauth_poly1305_amd64_constants:
-crypto_onetimeauth_poly1305_amd64_scale:
-.long 0x0,0x37f40000
-
-crypto_onetimeauth_poly1305_amd64_two32:
-.long 0x0,0x41f00000
-
-crypto_onetimeauth_poly1305_amd64_two64:
-.long 0x0,0x43f00000
-
-crypto_onetimeauth_poly1305_amd64_two96:
-.long 0x0,0x45f00000
-
-crypto_onetimeauth_poly1305_amd64_alpha32:
-.long 0x0,0x45e80000
-
-crypto_onetimeauth_poly1305_amd64_alpha64:
-.long 0x0,0x47e80000
-
-crypto_onetimeauth_poly1305_amd64_alpha96:
-.long 0x0,0x49e80000
-
-crypto_onetimeauth_poly1305_amd64_alpha130:
-.long 0x0,0x4c080000
-
-crypto_onetimeauth_poly1305_amd64_doffset0:
-.long 0x0,0x43300000
-
-crypto_onetimeauth_poly1305_amd64_doffset1:
-.long 0x0,0x45300000
-
-crypto_onetimeauth_poly1305_amd64_doffset2:
-.long 0x0,0x47300000
-
-crypto_onetimeauth_poly1305_amd64_doffset3:
-.long 0x0,0x49300000
-
-crypto_onetimeauth_poly1305_amd64_doffset3minustwo128:
-.long 0x0,0x492ffffe
-
-crypto_onetimeauth_poly1305_amd64_hoffset0:
-.long 0xfffffffb,0x43300001
-
-crypto_onetimeauth_poly1305_amd64_hoffset1:
-.long 0xfffffffe,0x45300001
-
-crypto_onetimeauth_poly1305_amd64_hoffset2:
-.long 0xfffffffe,0x47300001
-
-crypto_onetimeauth_poly1305_amd64_hoffset3:
-.long 0xfffffffe,0x49300003
-
-crypto_onetimeauth_poly1305_amd64_rounding:
-.byte 0x7f
-.byte 0x13
diff -u -r nacl-20100830.orig/okcompilers/c nacl-20100830/okcompilers/c
--- nacl-20100830.orig/okcompilers/c	2010-08-30 04:25:19.000000000 +0200
+++ nacl-20100830/okcompilers/c	2011-01-07 22:30:33.000000000 +0100
@@ -1,8 +1,8 @@
-gcc -m64 -fomit-frame-pointer
-gcc -m64 -O -fomit-frame-pointer
-gcc -m64 -O3 -fomit-frame-pointer -funroll-loops
-gcc -m32 -fomit-frame-pointer
-gcc -m32 -O -fomit-frame-pointer
-gcc -m32 -O3 -fomit-frame-pointer -funroll-loops
-spu-gcc -mstdmain -march=cell -O -fomit-frame-pointer -Drandom=rand -Dsrandom=srand
-spu-gcc -mstdmain -march=cell -O3 -funroll-loops -fomit-frame-pointer -Drandom=rand -Dsrandom=srand
+gcc -fPIC -m64 -fomit-frame-pointer
+gcc -fPIC -m64 -O -fomit-frame-pointer
+gcc -fPIC -m64 -O3 -fomit-frame-pointer -funroll-loops
+gcc -fPIC -m32 -fomit-frame-pointer
+gcc -fPIC -m32 -O -fomit-frame-pointer
+gcc -fPIC -m32 -O3 -fomit-frame-pointer -funroll-loops
+spu-gcc -fPIC -mstdmain -march=cell -O -fomit-frame-pointer -Drandom=rand -Dsrandom=srand
+spu-gcc -fPIC -mstdmain -march=cell -O3 -funroll-loops -fomit-frame-pointer -Drandom=rand -Dsrandom=srand
